src/handlers/v1/markets/[chainId]/[marketId]/zap-supply/index.ts

```diff
@@ -0,0 +1,251 @@
+      const _sourceToken = common.Token.from(sourceToken);
+      const _targetToken = common.Token.from(targetToken);
+
+      // 1. check supply collateral or base token
+      const targetCollateral = collaterals.find(({ asset }) => asset.is(_targetToken.unwrapped));
+      if (!targetCollateral && !_targetToken.is(baseToken)) {
+        throw newHttpError(400, { code: '400.5', message: 'target token is not collateral nor base' });
+      }
+
+      // 2. new and append swap token logic
+      const swapLogic = await newSwapTokenLogic(chainId, _sourceToken, amount, _targetToken, slippage);
+      if (swapLogic !== undefined) {
+        logics.push(swapLogic);
+        targetTokenAmount = swapLogic.fields.output.amount;
+      } else {
+        targetTokenAmount = amount;
+      }
+
+      // 3. new and append compound v3 supply logic
+      const supplyLogic = await newSupplyLogic(
+        chainId,
+        marketId,
+        _targetToken,
+        targetTokenAmount,
+        targetCollateral === undefined,
+        new BigNumberJS(borrowUSD)
+      );
+      logics.push(supplyLogic);
+
+      const estimateResult = await apisdk.estimateRouterData({ chainId, account, logics });
+      approvals = estimateResult.approvals;
+      permitData = estimateResult.permitData;
+
+      // 4. calc target position
+      targetPosition = calculateTargetPosition(targetCollateral, targetTokenAmount, marketInfo);
+    }
+
+    const responseBody: GetZapSupplyQuotationResponseBody = {
+      quotation: { targetTokenAmount, currentPosition, targetPosition },
+      approvals,
+      permitData,
+      logics,
+    };
+
+    return formatJSONResponse(responseBody);
+  },
+};
+
+async function newSwapTokenLogic(
+  chainId: number,
+  sourceToken: common.Token,
+  amount: string,
+  targetToken: common.Token,
+  slippage?: number
+) {
+  if (sourceToken.is(targetToken)) {
+    return undefined;
+  }
+
+  if (sourceToken.is(targetToken.unwrapped)) {
+    const wrapQuotation = await apisdk.protocols.utility.getWrappedNativeTokenQuotation(chainId, {
+      input: {
+        token: sourceToken,
+        amount,
+      },
+      tokenOut: targetToken,
+    });
+    return apisdk.protocols.utility.newWrappedNativeTokenLogic(wrapQuotation);
+  }
+
+  const quotation = await apisdk.protocols.paraswapv5.getSwapTokenQuotation(chainId, {
+    input: { token: sourceToken, amount },
+    tokenOut: targetToken,
+    slippage,
+  });
+  return apisdk.protocols.paraswapv5.newSwapTokenLogic(quotation);
+}
+
+async function newSupplyLogic(
+  chainId: number,
+  marketId: string,
+  token: common.Token,
+  amount: string,
+  isBase: boolean,
+  borrorwUSD: BigNumberJS
+) {
+  if (isBase) {
+    if (borrorwUSD.gt(0)) {
+      throw newHttpError(400, { code: '400.6', message: 'borrowUSD is not zero' });
+    }
+    const tokenList = await apisdk.protocols.compoundv3.getSupplyBaseTokenList(chainId);
+    const cToken = tokenList[marketId][0][1];
```

Comment:
Could use `marketInfo.baseToken`?

Reply:
If my understanding is correct, `baseToken` refers to USDC in the USDC market. But this is `cToken` (eg. cUSDCV3 in the USDC market).

Reply:
Yeah, I misunderstood. I found you've used `getCToken` in test, it it works?

```
const service = new logics.compoundv3.Service(chainId, hre.ethers.provider);
const cToken = await service.getCToken(marketId);
```

Reply:
Furthermore, if we can retrieve the cToken in the step 1, this would allow us to:
- Check 400.5 & 400.6 in step 1 for both cToken and collaterals.
- Check if targetToken.unwrapped is the cToken, allowing a more intuitive if-else in steps 3 & 4. Relying on `undefined` as a condition doesn't seem like a best practice to me."
- Simplify step 3 code without get cToken